src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -79,8 +100,35 @@ def extract_strings(self) -> Mapping[int, Set[str]]:
                 )
                 continue
 
+    @metrics.wraps("process_messages.filter_messages")
+    def filter_messages(self, keys_to_remove: Sequence[PartitionIdxOffset]) -> None:
+        for offset in keys_to_remove:
+            metrics.incr(
+                "sentry_metrics.indexer.process_messages.dropped_message",
+                tags={
+                    "reason": "cardinality_limit",
+                },
+            )
```

Comment:
Instead of incrementing this in the loop why not just count the length of `keys_to_remove` and increment with that number once.

Reply:
fixed, and also updated the set.add to be a set.update. i don't know what to do with the error log. we could choose to just not emit error logs, but tagging by organization could be a usecase in the future...